polyfactory/value_generators/complex_types.py

```diff
@@ -35,7 +35,13 @@ def handle_container_type(
     if isinstance(container, MutableMapping) or is_typeddict(container):
         key_type, value_type = unwrap_args(field_meta.annotation) or (str, str)
         key = handle_complex_type(field_meta=FieldMeta.from_type(annotation=key_type), factory=factory)
-        value = handle_complex_type(field_meta=FieldMeta.from_type(annotation=value_type), factory=factory)
+
+        if is_union(value_type) and field_meta.children and field_meta.children[0].children:
```

Comment:
Why would you need to have children[0].children? I would assume all unions would work the same here.

Reply:
I did like that because during debugging expected meta_fieds of the union were under field_meta.children[0].children but yep you're right it works also when just the children field is passed. I'm going to push without those indexes in a minute.

Reply:
I've uploaded the change :)

Reply:
And if you meant to not validate completely children's existence - without that pyright raises error so it's necessary at least like I pushed.

Error:  Argument of type "list[FieldMeta] | None" cannot be assigned to parameter "seq" of type "SupportsLenAndGetItem[_T@choice]" in function "choice"